### ISSUE TITLE
removing python requirement for systems with sshuttle preinstalled

### ISSUE THREAD

**Issue by axet** [sshuttle/sshuttle#888]
Hello!

On small systems like openwrt having python installed can be an issue (space, ram, cpu). Can we have no python requirement for systems where sshuttle installed as system package?

**Comment by brianmay** [comment]
Sorry, I think you lost me. sshuttle is a Python application, and needs Python, regardless of it is installed as a system package or not.

**Comment by axet** [comment]
I guess you right. But can you precompile it. Right?

**Comment by yjftsjthsd-g** [comment]
Not really; Python is an interpreted language. You could _try_ running it through nuitka or cython or such, but you're on your own with something that unusual (and if it even ran, there's no guarantee that it would use meaningfully less resources).

**Comment by DavHau** [comment]
@axet You can build a static python3 binary for arbitrary systems using [nix](https://nixos.org/):
```command
nix build nixpkgs#pkgsCross.aarch64-multiplatform.pkgsStatic.python3Minimal -o ./python
```
... and then copy it over to the openwrt machine's /tmp. It's only around 6 mb in size.
